lib/SILOptimizer/UtilityPasses/SILSkippingChecker.cpp

```diff
@@ -72,6 +72,15 @@ static bool shouldHaveSkippedFunction(const SILFunction &F) {
       return false;
   }
 
+  // Functions with @_backDeploy may be copied into the client, so they
+  // shouldn't be skipped. The SILFunction that may be copied into the client
+  // should be serialized and therefore is already handled above. However, a
+  // second resilient SILFunction is also emitted for back deployed functions.
+  // Since the body of the function as written was not skipped, it's expected
+  // that we see the SILFunction for the resilient copy here.
+  if (func->isBackDeployed())
```

Comment:
it seems a little bit odd for me that we need to check `BackDeployed` specifically here but not for other attributes like `@_alwaysEmitIntoClient`.

Reply:
`@_backDeployed` is special for the reasons described in the comment and in the PR description. Maybe there's a way I can describe this in a clearer way. `@inlinable`, AEIC, `@_transparent` are all handled by the `if (F.isSerialized())` check at the top of this function. `@_backDeployed` causes _two_ SILFunctions to be emitted. One of them passes the `if (F.isSerialized())` check. The other (representing the resilient copy of the function) is the one we need to handle here.

Reply:
ah, makes sense. Thank you for the clarification!